src/ProjectService.Business/Commands/Project/FindProjectsCommand.cs

```diff
@@ -10,87 +17,102 @@
 
-            List<Guid> departmentIds = dbProjects.Where(p => p.DepartmentId.HasValue).Select(p => p.DepartmentId.Value).ToList();
+      if (!departmentIds.Any())
+      {
+        return null;
+      }
 
-            if (!departmentIds.Any())
-            {
-                return null;
-            }
+      string errorMessage = "Cannot get departments now. Please try again later.";
 
-            string errorMessage = "Cannot get departments now. Please try again later.";
+      try
+      {
+        Response<IOperationResult<IGetDepartmentsResponse>> response = _rcGetDepartments
+          .GetResponse<IOperationResult<IGetDepartmentsResponse>>(
+            IGetDepartmentsRequest.CreateObj(departmentIds)).Result;
 
-            try
-            {
-                Response<IOperationResult<IGetDepartmentsResponse>> response = _rcGetDepartments
-                    .GetResponse<IOperationResult<IGetDepartmentsResponse>>(
-                        IGetDepartmentsRequest.CreateObj(departmentIds)).Result;
+        if (response.Message.IsSuccess)
+        {
+          return response.Message.Body.Departments;
+        }
 
-                if (response.Message.IsSuccess)
-                {
-                    return response.Message.Body.Departments;
-                }
+        _logger.LogWarning(string.Join(", ", response.Message.Errors));
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, errorMessage);
+      }
 
-                _logger.LogWarning(string.Join(", ", response.Message.Errors));
-            }
-            catch (Exception exc)
-            {
-                _logger.LogError(exc, errorMessage);
-            }
+      errors.Add(errorMessage);
 
-            errors.Add(errorMessage);
+      return null;
+    }
 
-            return null;
-        }
+    public FindProjectsCommand(
+      ILogger<FindProjectsCommand> logger,
+      IProjectRepository repository,
+      IFindProjectsResponseMapper responseMapper,
+      IRequestClient<IGetDepartmentsRequest> rcGetDepartments,
+      IHttpContextAccessor httpContextAccessor,
+      IBaseFindRequestValidator findRequestValidator)
+    {
+      _logger = logger;
+      _repository = repository;
+      _responseMapper = responseMapper;
+      _rcGetDepartments = rcGetDepartments;
+      _httpContextAccessor = httpContextAccessor;
+      _findRequestValidator = findRequestValidator;
+    }
 
-        public FindProjectsCommand(
-            ILogger<FindProjectsCommand> logger,
-            IProjectRepository repository,
-            IFindProjectsResponseMapper responseMapper,
-            IRequestClient<IGetDepartmentsRequest> rcGetDepartments)
-        {
-            _logger = logger;
-            _repository = repository;
-            _responseMapper = responseMapper;
-            _rcGetDepartments = rcGetDepartments;
-        }
+    public FindResponse<ProjectInfo> Execute(FindProjectsFilter filter)
```

Comment:
change to FindResultResponse from kernel

Reply:
done